src/EventListener/ReadListener.php

```diff
@@ -103,4 +112,30 @@ private function getItemData(Request $request, array $attributes)
 
         return $data;
     }
+
+    /**
+     * Gets data for a nested operation.
+     *
+     * @param Request $request
+     * @param array   $attributes
+     *
+     * @throws NotFoundHttpException
+     *
+     * @return object|null
+     */
+    private function getSubcollectionData(Request $request, array $attributes)
+    {
+        if (null === $this->subcollectionDataProvider) {
+            throw new RuntimeException('No subcollection data provider.');
+        }
```

Comment:
@dunglas should I throw this here? 

In ApiLoader I just fail silently but here I wasn't sure.